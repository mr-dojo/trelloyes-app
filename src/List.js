import React from 'react';
import Card from './Card';
import './List.css';

export default class List extends React.Component {
  static defaultProps = {
    header: '',
    cards: [],
  }
  render() {
    return (
      <section className='List'>
        <header className='List-header'>
          {this.props.header}
        </header>
        <div className='List-cards'>
          {this.props.cards.map((card) => 
            <Card
              key={card.id}
              title={card.title}
              content={card.content}
              />
          )}
          <button 
            type='button'
            onClick={() => this.props.onCreateRandom()}
          >
            + Add Random Card
          </button> 
        </div>
      </section>
    )
  }
}