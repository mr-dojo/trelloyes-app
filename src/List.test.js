import React from 'react';
import List from './List';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

const listComp = <List header='Test header' id='1' cards={[]}  />

describe('List componenet', () => {
  it ('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(listComp, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it ('Renders the UI with no changes', () => {
    const tree = renderer
      .create(listComp)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});