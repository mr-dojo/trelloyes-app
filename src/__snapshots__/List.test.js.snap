// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List componenet Renders the UI with no changes 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    Test header
  </header>
  <div
    className="List-cards"
  >
    <button
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
